experiments/functional/cstor-csi/csi-snapshot/test.yml

```diff
@@ -0,0 +1,124 @@
+- hosts: localhost
+  connection: local
+
+  vars_files:
+    - test_vars.yml
+    - /mnt/parameters.yml
+    
+  tasks:
+    - block:
+
+        ## Generating the testname for deployment
+        - include_tasks: /utils/fcm/create_testname.yml
+
+        ## RECORD START-OF-TEST IN LITMUS RESULT CR
+        - include_tasks: "/utils/fcm/update_litmus_result_resource.yml"
+          vars:
+            status: 'SOT'
+        
+        - name: Get the application pod name
+          shell: >
+            kubectl get po -n {{ namespace }} --no-headers -o custom-columns=:.metadata.name
+          args:
+            executable: /bin/bash
+          register: app_pod_name
+
+        - name: Check if the application pod is in running state
+          shell: >
+            kubectl get pods -n {{ namespace }} --no-headers -o custom-columns=:.status.phase
+          args:
+            executable: /bin/bash
+          register: pod_status
+          failed_when: "'Running' not in pod_status.stdout"
+
+        - name: Create some test data
+          include_tasks: "/utils/scm/applications/busybox/busybox_data_persistence.yml"
+          vars:
+            status: 'LOAD'
+            ns: "{{ namespace }}"
+            pod_name: "{{ app_pod_name.stdout }}"  
+          when: data_persistence != ''
+
+        - name: generating yaml file for snapshot
+          template: 
+            src: snapshot.j2
+            dest: snapshot.yml
+
+        - name: create snapshot    
+          shell: >
+            kubectl create -f snapshot.yml -n "{{ namespace }}"
+          args:
+            executable: /bin/bash
+
+       #  Verify if the snapshot created successfully
+        - name: Obtaining the storage class used from PVC
+          shell: kubectl get pvc {{ pvc_name }} -n {{ namespace }} --no-headers -o custom-columns=:.spec.storageClassName
+          args:
+            executable: /bin/bash
+          register: app_sc
+
+        - name: Obtaining the SPC from storage class
+          shell: kubectl get sc {{ app_sc.stdout }} --no-headers  -o yaml | awk '/StoragePoolClaim/{getline; print}' | cut -d ':' -f2 | sed 's/"//g'
```

Comment:
csi based volume can be created only on cspc not spc

Reply:
ok, resolved